src/libsync/networkjobs.cpp

```diff
@@ -834,52 +766,84 @@ bool DetermineAuthTypeJob::finished()
     return true;
 }
 
-SimpleNetworkJob::SimpleNetworkJob(AccountPtr account, QObject *parent)
-    : AbstractNetworkJob(account, QString(), parent)
+SimpleNetworkJob::SimpleNetworkJob(AccountPtr account, const QString &path, const QByteArray &verb, const QNetworkRequest &req, QObject *parent)
+    : AbstractNetworkJob(account, path, parent)
+    , _verb(verb)
+    , _request(req)
 {
 }
 
-void SimpleNetworkJob::start()
+SimpleNetworkJob::SimpleNetworkJob(AccountPtr account, const QString &path, const QByteArray &verb, const UrlQuery &arguments, const QNetworkRequest &req, QObject *parent)
+    : SimpleNetworkJob(account, path, verb, req, parent)
 {
-    sendRequest(_simpleVerb, _simpleUrl, _simpleRequest, _simpleBody);
-    AbstractNetworkJob::start();
+    if (verb == QByteArrayLiteral("GET")) {
+        // TODO: any better idea?
```

Comment:
You mentioned you want to use an enum instead of strings for `verb`. I'd put that into the comment.

Reply:
I was referencing toPercentEncoding on the arguments.

Reply:
I guess that comment should be moved there then, or you should mention that in the comment. When the class will be revisited by someone else, they might get confused as well.

Reply:
The todo was meant to start a discussion in the review and not to stay there :)